appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java

```diff
@@ -465,7 +457,7 @@
     // an EJB Remote interface to be a subtype of the Service Endpoint
     // Interface.  In that case, it's ambiguous to do a lookup based only
     // on a java.lang.reflect.Method
-    protected Map webServiceInvocationInfoMap = new HashMap();
+    protected Map<Method, InvocationInfo> webServiceInvocationInfoMap = new HashMap<>();
```

Comment:
Now I see, it should be final. But it is protected, so Eclipse was afraid that it could break something. It could, but not in Payara.

Reply:
True, there is too much synchronization (or none here). But I'm afraid of that, TCK is sequential and slow, but I'm not sure with applications with thousands of threads. The code is not much well prepared for concurrency and we don't have much tests for that.
I will try to create some later ...